scripts/process_vad_data.py

```diff
@@ -0,0 +1,333 @@
+                current_offset = 0.0
+
+                while current_offset < duration:
+                    difference = duration - current_offset
+                    segment_duration = min(duration_max, difference)
+
+                    offsets.append(current_offset)
+                    durations.append(segment_duration)
+
+                    current_offset += duration_stride
+
+            else:
+                offsets.append(0.0)
+                durations.append(duration)
+
+            for duration, offset in zip(durations, offsets):
+                metadata = {
+                    'audio_filepath': file,
+                    'duration': duration,
+                    'label': label,
+                    'text': '_',  # for compatibility with ASRAudioText
+                    'offset': offset,
+                }
+                json.dump(metadata, fout)
+                fout.write('\n')
+                fout.flush()
+                seg_num += 1
+    return skip_num, seg_num, output_path
+
+
+def load_list_write_manifest(
+    data_dir, out_dir, filename, prefix, duration_stride=1.0, duration_max=1.0, duration_limit=100.0, filter_long=True
+):
+
+    filename = prefix + '_' + filename
+    file_path = os.path.join(data_dir, filename)
+
+    with open(file_path, 'r') as allfile:
+        files = allfile.read().splitlines()
+
+    manifest_name = filename.split('_list.txt')[0] + '_manifest'
+    skip_num, seg_num, output_path = write_manifest(
+        data_dir,
+        out_dir,
+        files,
+        prefix,
+        manifest_name,
+        duration_stride,
+        duration_max,
+        duration_limit,
+        filter_long=True,
+    )
+    return skip_num, seg_num, output_path
+
+
+def get_max_json(data_dir, data_json, max_limit, prefix):
+    data = []
+    seg = 0
+    for line in open(data_json, 'r'):
+        data.append(json.loads(line))
+    filename = data_json.split('/')[-1]
+
+    fout_path = os.path.join(data_dir, prefix + "_" + filename)
+    selected_sample = random.sample(data, max_limit)
+
+    with open(fout_path, 'a') as fout:
+        for i in selected_sample:
+            seg += 1
+            json.dump(i, fout)
+            fout.write('\n')
+            fout.flush()
+
+    logging.info(f'Get {seg}/{max_limit} to  {fout_path} from {data_json}')
+    return fout_path
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Speech and backgound data download and preprocess')
+    parser.add_argument("--out_dir", required=True, default='./manifest/', type=str)
+    parser.add_argument("--speech_data_root", required=True, default=None, type=str)
+    parser.add_argument("--background_data_root", required=True, default=None, type=str)
+    parser.add_argument('--test_size', required=False, default=0.1, type=float)
+    parser.add_argument('--val_size', required=False, default=0.1, type=float)
+    parser.add_argument('--log', required=False)
+    parser.add_argument('--rebalance', required=False)
```

Comment:
This needs to have `action='store_true'` after `required=False`.

Reply:
Good catch! fixed